### PR TITLE
Adapt batch update strategy

### PR DESCRIPTION
**Pull request by imbajin**

### CODE REVIEW

File: src/test/java/com/baidu/hugegraph/api/BatchUpdateElementApiTest.java [hunk 1]
```diff
@@ -173,6 +173,28 @@ public void testVertexBatchUpdateStrategyEliminate() {
...
 
+    @Test
+    public void testVertexBatchUpdateStrategyOverride() {
+        BatchVertexRequest req = batchVertexRequest("price", -1, 1,
+                                                    UpdateStrategy.OVERRIDE);
+
+        List<Vertex> vertices = vertexAPI.update(req);
+        assertBatchResponse(vertices, "price", 1);
+
+        req = batchVertexRequest("list", "old", "new", UpdateStrategy.OVERRIDE);
```
Comment (javeme) [inline src/test/java/com/baidu/hugegraph/api/BatchUpdateElementApiTest.java]: there is always only one element in the list, is it expected and reasonable?

Comment (imbajin) [reply src/test/java/com/baidu/hugegraph/api/BatchUpdateElementApiTest.java]: `OVERRIDE`  is used for this case:
> Person have 3 props: `name, price, list` ,and **only** `price` have update strategy
> 1. Old value from backend : `tom, -1, [oldStr1, oldStr2](list)` 
> 2. New value1 from json : `tom, 1, [newStr1, newStr2]`
> 3. New value2 from json : `tom, null, null`
> 4. After update , get: `tom, 1, [newStr1, newStr2]`

however, if we don't set update strategy for list, we will get `tom, 1`  and lose the `list` property.

Comment (javeme) [reply src/test/java/com/baidu/hugegraph/api/BatchUpdateElementApiTest.java]: That's OK,  if we can override the list as a whole.

File: src/main/java/com/baidu/hugegraph/api/graph/structure/BatchEdgeRequest.java [hunk 2]
```diff
@@ -45,6 +45,10 @@ public BatchEdgeRequest() {
         this.createIfNotExist = true;
```
Comment (zhoney) [inline src/main/java/com/baidu/hugegraph/api/graph/structure/BatchEdgeRequest.java]: Expand accepted api version(0.45) in HugeClient

File: src/main/java/com/baidu/hugegraph/driver/HugeClient.java [hunk 3]
```diff
@@ -120,7 +120,7 @@ private void initManagers(RestClient client, String graph) {
     private void checkServerApiVersion() {
         VersionUtil.Version apiVersion = VersionUtil.Version.of(
                                          this.version.getApiVersion());
-        VersionUtil.check(apiVersion, "0.38", "0.45",
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/driver/HugeClient.java]: upgrade pom version to 1.7.7

File: src/main/java/com/baidu/hugegraph/structure/schema/VertexLabel.java [hunk 4]
```diff
@@ -78,7 +78,7 @@ public String toString() {
 
         Builder useCustomizeNumberId();
 
-        Builder useCustomizeUuid();
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/structure/schema/VertexLabel.java]: also update IdStrategy.isCustomize() method to 
```
this == IdStrategy.CUSTOMIZE_STRING || this == IdStrategy.CUSTOMIZE_NUMBER || this == IdStrategy.CUSTOMIZE_UUID;

```

and please add method in IdStrategy.class

```
public boolean isCustomizeUUID() {
        return this == IdStrategy.CUSTOMIZE_UUID;
    }
```

File: src/main/java/com/baidu/hugegraph/structure/schema/VertexLabel.java [hunk 5]
```diff
@@ -78,7 +78,7 @@ public String toString() {
...
+        Builder useCustomizeUUID();
 
         Builder properties(String... properties);
```
Comment (Linary) [inline src/main/java/com/baidu/hugegraph/structure/schema/VertexLabel.java]: also please add method in DataType.class
```
public boolean isBoolean() {
        return this == BOOLEAN;
    }
```

File: src/main/java/com/baidu/hugegraph/driver/HugeClient.java [hunk 6]
```diff
@@ -120,7 +120,7 @@ private void initManagers(RestClient client, String graph) {
...
+        VersionUtil.check(apiVersion, "0.38", "0.46",
```
Comment (javeme) [inline src/main/java/com/baidu/hugegraph/driver/HugeClient.java]: check no UpdateStrategy.OVERRIDE if apiVersion<0.45 at sending request

**Pull request merged**